Import example dataset

ewang12 opened the pull request:
This script runs after the Visual Insights trial is set up, and uses the CLI to upload a Bowls_and_Plates.zip example dataset.

Some questions:
1) Referenced the fetch_vision.sh part for using aria2 to fetch Bowls_and_plates.zip from COS. Not sure if it was better to keep it all contained here and reinstall aria2 again to fetch Bowls_and_plate.zip here, or include it in fetch_vision.sh where you're already installing aria2 and just grab it there. 

2) Should the github repo in line 34 be passed in as a variable? Or is it okay to have it as a a direct URL? 

@mikehollinger @atthorst

mikehollinger commented on scripts/import_dataset.sh:
```diff
@@ -0,0 +1,76 @@
+apt-get -o Dpkg::Use-Pty=0 update -qq  || echo " RC${?} Got an error on update???"
+apt-get -o Dpkg::Use-Pty=0 install -qq aria2
+
+cd $HOME
```
Hey! don't leave stuff in $HOME; instead use /tmp and tidy up after yourself at the end of the script by deleting the file.

ewang12 replied:
Thanks for the comments! I wasn't sure if it would be better to leave the file for them there in case they wanted to re-upload it the example dataset again at some point to play around with the dataset and then maybe expand the comment to include that information? Not sure what the typical use case would be/how long they would have the instance up.

mikehollinger commented on scripts/import_dataset.sh:
```diff
@@ -0,0 +1,76 @@
+BASEDIR="$(dirname "$0")"
+source ${BASEDIR}/env.sh
+
+# Install aria and get example dataset zip file
```
This is such a small file that installing aria, downloading it, and uninstalling aria takes longer than just wget'ing it. So delete all the aria stuff from line 19 to 23

ewang12 replied:
Ah okay, thanks! Will update to use wget and output the file as a set name to eliminate the need for that DATASET_NAME variable as well.

mikehollinger commented on scripts/import_dataset.sh:
```diff
@@ -0,0 +1,76 @@
+
+cd $HOME
+echo "Fetching example dataset from ${URLPAIVDATASET}"
+aria2c -q $URLPAIVDATASET
```
switch to a simple wget or curl (with appropriate flags) to output the file where you want it.

mikehollinger commented on scripts/import_dataset.sh:
```diff
@@ -0,0 +1,76 @@
+aria2c -q $URLPAIVDATASET
+
+# Uninstall aria2
+echo "Uninstalling aria2"
```
You can delete the aria package management stuff here...

mikehollinger commented on scripts/import_dataset.sh:
```diff
@@ -0,0 +1,76 @@
+
+echo "SUCCESS: Downloaded ${DATASET_NAME} example data set successfully."
+
+git clone https://github.com/IBM/vision-tools.git
```
i'm surprised git is present on the system. :-)

ewang12 replied:
haha yeah! I thought I would have to install it but turns out it's already there

mikehollinger commented on scripts/import_dataset.sh:
```diff
@@ -0,0 +1,76 @@
+
+
+# Import Bowls_and_Plates.zip data set
+vision datasets import $HOME/${DATASET_NAME}
```
change to wherever this file is (see my earlier comment about $HOME

mikehollinger commented on vm.tf:
```diff
@@ -38,6 +38,11 @@ variable "vision_tar_name" {
 }
 
+variable "example_dataset_name" {
+  description = "Name of example dataset to automatically import into Visual Insights."
```
I'm torn here; should this be relative to our bucket base, or should it be an absolute URL? On the one hand, having an absolute URL will be super-handy for ppl who want to set up some sort of class; i.e. they'll be able to point to some random COS bucket or URL and say "here, use this," and it'll get prepopulated in. On the other hand, that now means we'll have two similar-ish but different ways to handle injecting assets into the the VM. What do you think about changing this to a fully-specified URL, ala "https://foo/bar/baz.zip" ?

ewang12 replied:
I think it would work out to change it to a fully-specific URL! Was mainly basing it just on the other variables in the schematic, but would also make sense to just have it a full URL as well and for other people to easily specify their own file.

mikehollinger commented on vm.tf:
```diff
@@ -210,6 +215,8 @@ export USERMGTIMAGE=vision-usermgt:${var.vision_version}
 export COS_BUCKET_BASE=${var.cos_bucket_base}
 export URLPAIVIMAGES="$${COS_BUCKET_BASE}/${var.vision_tar_name}"
 export URLPAIVDEB="$${COS_BUCKET_BASE}/${var.vision_deb_name}"
+export URLPAIVDATASET="$${COS_BUCKET_BASE}/${var.example_dataset_name}"
```
see my earlier comment about the URL...

mikehollinger commented on vm.tf:
```diff
@@ -210,6 +215,8 @@ export USERMGTIMAGE=vision-usermgt:${var.vision_version}
 export URLPAIVIMAGES="$${COS_BUCKET_BASE}/${var.vision_tar_name}"
 export URLPAIVDEB="$${COS_BUCKET_BASE}/${var.vision_deb_name}"
+export URLPAIVDATASET="$${COS_BUCKET_BASE}/${var.example_dataset_name}"
+export DATASET_NAME="${var.example_dataset_name}"
```
This takes a couple of args, and also uses two different environment variables (`URLPAIVDATASET` and `DATASET_NAME`). If you choose to make the change to make the sample dataset a full URL, then this DATASET_NAME variable will be different. You don't really need the name here; why not simply save the file explicitly as "dataset.zip" with the appropriate flag for wget or curl, and then hard-code that as a known constant in the download bash script? That means you won't need this in the constants.

mikehollinger commented on scripts/import_dataset.sh:
```diff
@@ -0,0 +1,76 @@
+
+
+# Import Bowls_and_Plates.zip data set
+vision datasets import $HOME/${DATASET_NAME}
```
You can change this to NOT need the DATASET_NAME and instead use a hardcoded value by changing the wget/curl/aria command to output the file as "/whereveryouwant/whateveryouwant.zip" explicitly.

ewang12 replied:
Thanks, updating to wget to output as a set name for the dataset so this variable is no longer needed

ewang12 commented:
Hi @mikehollinger 
Thanks for the review!
Made the following changes and tested by applying the plan and confirming dataset was uploaded and directory removed at the end:

- Moved code from running in $HOME to /tmp/load_dataset
- Used wget to download file instead of aria and output as set file name to eliminate need of dataset name variable 
- Changed the variable in vm.tf to take a URL instead of the dataset name
- Remove and clean up the /tmp/load_dataset directory at the end
